.github/workflows/release-please.yml

```diff
@@ -7,44 +7,87 @@ on:
 
     permissions:
       contents: write # Contents and pull-requests are for release-please to make releases.
       pull-requests: write
 
+    outputs:
+      release_created: ${{ steps.release.outputs.release_created }}
+      prs_created: ${{ steps.release.outputs.prs_created }}
+      pr_branch_name: ${{ steps.release.outputs.prs_created == 'true' && fromJSON(steps.release.outputs.pr).headBranchName || '' }}
     steps:
       - uses: google-github-actions/release-please-action@v4
         id: release
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
 
+  update-release-pr:
+    needs: release-please
+    if: ${{ needs.release-please.outputs.prs_created == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
       - uses: actions/checkout@v4
-
       - name: Update launchdarkly-server-sdk rockspec
         uses: ./.github/actions/update-versions
-        if: ${{ steps.release.outputs.prs_created == 'true' }}
         with:
           package: launchdarkly-server-sdk
-          branch: ${{ fromJSON(steps.release.outputs.pr).headBranchName }}
+          branch: ${{ needs.release-please.outputs.pr_branch_name }}
 
       - name: Update launchdarkly-server-sdk-redis rockspec
         uses: ./.github/actions/update-versions
-        if: ${{ steps.release.outputs.prs_created == 'true' }}
         with:
           package: launchdarkly-server-sdk-redis
-          branch: ${{ fromJSON(steps.release.outputs.pr).headBranchName }}
-
-      - name: Build and Test
-        if: ${{ steps.release.outputs.release_created == 'true' }}
-        uses: ./.github/actions/ci
+          branch: ${{ needs.release-please.outputs.pr_branch_name }}
 
+  publish-docs:
+    needs: release-please
+    if: ${{ needs.release-please.outputs.release_created == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
       - name: Build documentation
-        if: ${{ steps.release.outputs.release_created == 'true' }}
         uses: ./.github/actions/build-docs
 
-      - uses: ./.github/actions/publish-docs
-        if: ${{ steps.release.outputs.release_created == 'true' }}
+      - name: Publish docs
+        uses: ./.github/actions/publish-docs
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+
+  publish-server:
+    needs: [release-please, rockspec-info]
+    if: ${{ needs.release-please.outputs.release_created == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - uses: launchdarkly/gh-actions/actions/release-secrets@release-secrets-v1.0.2
+      name: 'Get LuaRocks token'
+      with:
+        aws_assume_role: ${{ vars.AWS_ROLE_ARN }}
+        ssm_parameter_pairs: '/production/common/releasing/luarocks/token = LUAROCKS_API_TOKEN'
+    - uses: ./.github/actions/publish
+      name: Publish server package
+      with:
+        dry_run: 'false'
+        rockspec: ${{ fromJSON(needs.rockspec-info.outputs.info).server }}
+
+  publish-redis:
+    needs: [ publish-server, rockspec-info ]
```

Comment:
I assume redis never needs published without server needing published?

Reply:
(Github really needs to add a better "run jobs 1 at a time" setup to make life just simpler. Ideally js-core would publish in order, but doing so with tons of conditional releases is annoying.)

Reply:
Currently release please is only aware of the root package. I couldn't think of any compelling reason to publish one without the other, and didn't want to change the directory layout.

In retrospect, a compelling reason is just clarity. Will probably split it into two packages like C++ but needs a bit more thought.